The version of your git module can change your default branch between main and master and breaks unit tests

Currently when running with a git module that defaults to initial branch of main (this is what I get on a fresh install on a mac with python 3.12 and git 3.1.42) you get:

.pkg: _optional_hooks> python /Users/dmcphers/Code/cli/venv/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /Users/dmcphers/Code/cli/venv/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_wheel> python /Users/dmcphers/Code/cli/venv/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /Users/dmcphers/Code/cli/venv/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /Users/dmcphers/Code/cli/venv/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
unit: install_package> python -I -m pip install --force-reinstall --no-deps /Users/dmcphers/Code/cli/.tox/.tmp/package/329/cli-0.0.1.tar.gz
unit: commands[0]> pytest tests
================================= test session starts =================================
platform darwin -- Python 3.12.2, pytest-8.1.1, pluggy-1.4.0
cachedir: .tox/unit/.pytest_cache
rootdir: /Users/dmcphers/Code/cli
configfile: pyproject.toml
plugins: anyio-4.3.0
collected 30 items                                                                    

tests/test_config.py ...                                                        [ 10%]
tests/test_lab.py .                                                             [ 13%]
tests/test_lab_download.py ..                                                   [ 20%]
tests/test_lab_init.py .......                                                  [ 43%]
tests/test_lab_list.py FF...                                                    [ 60%]
tests/test_lab_train.py ...........                                             [ 96%]
tests/test_notebooks.py .                                                       [100%]

====================================== FAILURES =======================================
________________________________ TestLabList.test_list ________________________________

self = <tests.test_lab_list.TestLabList testMethod=test_list>

    def test_list(self):
        untracked_file = "compositional_skills/new/qna.yaml"
        tracked_file = "compositional_skills/tracked/qna.yaml"
        self.taxonomy.create_untracked(untracked_file)
        self.taxonomy.add_tracked(tracked_file)
        runner = CliRunner()
        with runner.isolated_filesystem():
            result = runner.invoke(
                lab.list,
                [
                    "--taxonomy-base",
                    TAXONOMY_BASE,
                    "--taxonomy-path",
                    self.taxonomy.root,
                ],
            )
>           self.assertIn(untracked_file, result.output)
E           AssertionError: 'compositional_skills/new/qna.yaml' not found in 'Generating dataset failed with the following error: Couldn\'t find the taxonomy base branch "master" from the current HEAD\n'

tests/test_lab_list.py:40: AssertionError
_________________________ TestLabList.test_list_invalid_base __________________________

self = <tests.test_lab_list.TestLabList testMethod=test_list_invalid_base>

    def test_list_invalid_base(self):
        taxonomy_base = "main"
        runner = CliRunner()
        with runner.isolated_filesystem():
            result = runner.invoke(
                lab.list,
                [
                    "--taxonomy-base",
                    taxonomy_base,
                    "--taxonomy-path",
                    self.taxonomy.root,
                ],
            )
            self.assertIsNone(result.exception)
>           self.assertIn(
                f'Couldn\'t find the taxonomy base branch "{taxonomy_base}" '
                "from the current HEAD",
                result.output,
            )
E           AssertionError: 'Couldn\'t find the taxonomy base branch "main" from the current HEAD' not found in ''

tests/test_lab_list.py:95: AssertionError
================================== warnings summary ===================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

.tox/unit/lib/python3.12/site-packages/pydantic/_internal/_fields.py:151
  /Users/dmcphers/Code/cli/.tox/unit/lib/python3.12/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_path" has conflict with protected namespace "model_".
  
  You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================== short test summary info ===============================
FAILED tests/test_lab_list.py::TestLabList::test_list - AssertionError: 'compositional_skills/new/qna.yaml' not found in 'Generating datas...
FAILED tests/test_lab_list.py::TestLabList::test_list_invalid_base - AssertionError: 'Couldn\'t find the taxonomy base branch "main" from the current H...
====================== 2 failed, 28 passed, 3 warnings in 2.71s =======================
unit: exit 1 (3.23 seconds) /Users/dmcphers/Code/cli> pytest tests pid=21403
  unit: FAIL code 1 (6.97=setup[3.74]+cmd[3.23] seconds)
  evaluation failed :( (7.00 seconds)

--- a/tests/taxonomy.py
+++ b/tests/taxonomy.py
@@ -12,7 +12,7 @@ class MockTaxonomy:
 
     def __init__(self, path: Path) -> None:
         self.root = path
-        self._repo = git.Repo.init(path)
+        self._repo = git.Repo.init(path, initial_branch="main")
         with open(path / self.INIT_COMMIT_FILE, "wb"):
             pass
         self._repo.index.add([self.INIT_COMMIT_FILE])

--- a/tests/test_lab_list.py
+++ b/tests/test_lab_list.py
@@ -8,7 +8,7 @@ import pytest
 # First Party
 from cli import lab
 
-TAXONOMY_BASE = "master"
+TAXONOMY_BASE = "main"
 
 
 class TestLabList(unittest.TestCase):
@@ -74,12 +74,12 @@ class TestLabList(unittest.TestCase):
                 ],
             )
             self.assertListEqual(self.taxonomy.untracked_files, [untracked_file])
-            # Invalid extension is silently filtered out.
+            # Invalid extension is silently filtered out
             self.assertNotIn(untracked_file, result.output)
             self.assertEqual(result.exit_code, 0)
 
     def test_list_invalid_base(self):
-        taxonomy_base = "main"
+        taxonomy_base = "invalid"
         runner = CliRunner()
         with runner.isolated_filesystem():
             result = runner.invoke(

